src/index.ts

```diff
@@ -203,16 +203,39 @@ type DeepChangedCache = WeakMap<object, {
 }>;
 
 /**
- * compare two object
+ * Compare changes on objects.
  *
- * It will compare only with affected object properties
+ * This will compare the affected properties on tracked objects inside the proxy
+ * to check if there were any changes made to it,
+ * by default if no property was accessed on the proxy it will attempt to do a
+ * reference equality check for the objects provided (Object.is(a, b)). If you access a property
+ * on the proxy, then isDeepChanged will only compare the affected properties.
+ *
+ * @param {object} obj - The original object to compare.
+ * @param {object} obj - Object to compare with the original one.
+ * @param {WeakMap<object, unknown>} affected -
+ * WeakMap that holds the tracking of which properties in the proxied object were accessed.
+ * @param {WeakMap<object, unknown>} [cache] -
+ * WeakMap that will help keep referential identity for proxies.
```

Comment:
This is different from `proxyCache`. It's for caching the comparison result, 1) for performance with repetitive comparisons, and 2) to avoid infinite loop with circular structure.

Reply:
Oh yeah, that makes sense. Will update 👍

Reply:
@dai-shi updated with your description.